include/boost/url/impl/url.ipp

```diff
@@ -2343,4 +2343,20 @@ operator<<(
 } // urls
 } // boost
 
+#ifdef BOOST_URL_HAS_STD_FORMAT
+template <>
+struct std::formatter<boost::urls::url>
+    : std::formatter<std::string_view>
+{
+    template <typename FormatContext>
+    auto format(
+        const boost::urls::url& u,
+        FormatContext& ctx)
+    {
+        return formatter<string_view>::format(
+            std::string_view(u.c_str(), u.size()), ctx);
```

Comment:
`u.string()`

Reply:
Also you should use `u.data()` instead of `u.c_str()` (if you wanted the pointer and size)